### DISCUSSION TITLE
model.forwad(x) is huge times faster than model(x)

### DISCUSSION THREAD

**Discussion by 0xmelanoleuca** [cornellius-gp/gpytorch#2002] [Q&A] [answered]
When I try to predict after training the ExactGP, using model(x) should take 10 hours for the whole Bayesian Optimization training. However, if I change model(x) into model.forward(), it will only spend 10 minutes. What is the difference between the two methods that makes such a huge difference in results？
The following is my predict function and model
    ```
def get_output(self, X, model, return_std=True):

        if not isinstance(X, torch.Tensor):
            X = torch.tensor(X)
        pred_y = model.forward(X.cuda())
        # pred_y = model.likelihood(pred_y)
        mu = pred_y.mean.detach().cpu().numpy()
        std = np.sqrt(pred_y.variance.detach().cpu().numpy())
        if return_std:
            return mu, std
        return mu
```
```
class GPModel(models.ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super(GPModel, self).__init__(train_x, train_y, likelihood)
        self.mean_module = means.ConstantMean()
        self.covar_module = kernels.ScaleKernel(kernels.RBFKernel())

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return distr.MultivariateNormal(mean_x, covar_x)
```

**Comment by wjmaddox** [comment, marked as answer]
This is because calling the forward directly is just evaluating the GP prior, while in eval mode, you are computing the GP posterior distribution. 

This distinction is explained here: https://docs.gpytorch.ai/en/stable/examples/01_Exact_GPs/Simple_GP_Regression.html#Make-predictions-with-the-model
